pkg/master/pod_cache.go

```diff
@@ -192,6 +192,7 @@ func (p *PodCache) computePodStatus(pod *api.Pod) (api.PodStatus, error) {
 	} else {
 		newStatus.Info = result.Status.Info
 		newStatus.PodIP = result.Status.PodIP
+		newStatus.Conditions = result.Status.Conditions
```

Comment:
See #4188. We'd like the last known status to be returned to the user. Once we add LastTransitionTime and LastProbeTime, similar to NodeCondition, clients will be able to interpret that. PodUnknown is far less useful and more disruptive. Also, we don't want application availability to necessarily be coupled to Kubelet availability, so blacking out the application when Kubelet doesn't respond isn't the right thing. We will, however, need to improve the signals with respect to node reachability.

What I expect readiness to help the most with respect to load balancing is planned outages. I also expect to use it in numerous disruptive automation processes, such as rolling updates, auto-scaling, node draining, re-scheduling, etc.